src/gurt/hash.c

```diff
@@ -1265,6 +1265,10 @@ d_hhash_link_lookup(struct d_hhash *hhash, uint64_t key)
 	if (d_hhash_key_isptr(key)) {
 		struct d_hlink *hlink = (struct d_hlink *)key;
 
+		if (hlink == NULL) {
+			D_ERROR("NULL PTR type key.\n");
+			return NULL;
+		}
```

Comment:
Is this change intentional or left-over debugging?

Reply:
This is for robust error handling. Instead of crashing it should print an error message.